### DISCUSSION TITLE
Фильтр разнородного списка

### DISCUSSION THREAD

**Discussion by erpic-ru** [totumonline/totum-ru-issues-and-discussions#160] [Как это сделать?] [answered]
Telegram присылает массив с разным набором ключей. При попытке разбора возникает ошибка (ключ не найден).

Подскажите, пожалуйста, как при помощи listfilter или ещё чего-то получить только список значений, в которых есть ключ "message", внутри которого есть поле "message_id"?

<img width="890" alt="Снимок экрана 2022-06-22 в 09 20 03" src="https://user-images.githubusercontent.com/91674114/174948693-3ff5dc15-0700-4749-afc5-57b113d3c878.png">

**Comment by erpic-ru** [comment, marked as answer]
Добавить параметр skip: true в listfilter

**Reply by ttm-alexeypolunin** [reply]
А можешь прикрепить, как ты сделал фильтрацию по внутреннему `message_id`, тк `listFilter` только на первом уровне работает. Я в таких случаях вытаскивал параметр наверх через `listReplace`.

**Reply by erpic-ru** [reply]
```
=: listFilter(list: $1[[message]]; key: "message_id" != 0 ;  skip: true)


1:listFilter(list: #dannye; key: "message" != 0 ;  skip: true)
```
